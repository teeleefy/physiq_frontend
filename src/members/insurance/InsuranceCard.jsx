import {
    Card,
    CardBody,
    CardText,
    ListGroupItem
  } from "reactstrap";
  import { NavLink } from "react-router-dom";
  import '../styles/Insurance.css'


function Insurance({insurance, memberId}){

    function renderInsuredName(){
        if(insurance.insuredName){
            return (
                <>
                 <p><b>Insured Name:</b> {insurance.insuredName}</p>
                </>
            )
        }
    }

    function renderStartDate(){
        if(insurance.startDate){
            return (
                <>
                 <p><b>Start Date:</b> {insurance.startDate}</p>
                </>
            )
        }
    }

    function renderEndDate(){
        if(insurance.endDate){
            return (
                <>
                 <p><b>End Date:</b> {insurance.endDate}</p>
                </>
            )
        }
    }

    function renderGroupNum(){
        if(insurance.groupNum){
            return (
                <>
                 <p><b>Group Number:</b> {insurance.groupNum}</p>
                </>
            )
        }
    }

    function renderContractNum(){
        if(insurance.contractNum){
            return (
                <>
                 <p><b>Contract Number:</b> {insurance.contractNum}</p>
                </>
            )
        }
    }

    function renderNotes(){
        if(insurance.notes){
            return (
                <>
                 <p><b>Notes:</b> {insurance.notes}</p>
                </>
            )
        }
    }

    return(
        <>
            <ListGroupItem >
                <Card className="Insurance-card">
                    <CardBody>
                        <CardText>
                            <p><b>Insurance Name:</b> <NavLink className="InsuranceCard-name" to={`/member/${memberId}/insurance/${insurance.id}`} key={insurance.id}>
                                {insurance.companyName}
                            </NavLink></p>
                            <p><b>Insurance Type:</b> {insurance.type}</p>
                            {renderInsuredName()}
                            {renderStartDate()}
                            {renderEndDate()}
                            {renderGroupNum()}
                            {renderContractNum()}
                            {renderNotes()}
                        </CardText>
                    </CardBody>
                </Card>
            </ListGroupItem>
        </>
     )
}

export default Insurance;