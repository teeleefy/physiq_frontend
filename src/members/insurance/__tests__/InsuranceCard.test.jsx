import { renderWithRouter, screen } from '../../../../__tests__/test-tools';
import InsuranceCard from '../InsuranceCard'

let insurance={type: 'Medical', company_name: "Medicare", insured_name: "John K. Smith", start_date: "2024-01-01", end_date: "2024-12-31", group_num: "123456", contract_num: "DSA9888DSSFI", notes: "Free health insurance. Remember to fill out paperwork again in November to reapply for free program." }
let memberId=1000;

describe('Insurance', () => {
  it('renders the Insurance component', () => {
    renderWithRouter(<InsuranceCard insurance={insurance} memberId={memberId}/>)
    
    screen.debug(); // prints out the jsx in the Insurance component unto the command line
  })
})

it("matches snapshot", function() {
  const { asFragment } = renderWithRouter(<InsuranceCard insurance={insurance} memberId={memberId}/>);
  expect(asFragment()).toMatchSnapshot();
});