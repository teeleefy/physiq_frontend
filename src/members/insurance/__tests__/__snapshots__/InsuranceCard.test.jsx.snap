// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="list-group-item"
  >
    <div
      class="Insurance-card card"
    >
      <div
        class="card-body"
      >
        <p
          class="card-text"
        />
        <p>
          <b>
            Insurance Name:
          </b>
           
          <a
            class="InsuranceCard-name"
            href="/member/1000/insurance/undefined"
          />
        </p>
        <p>
          <b>
            Insurance Type:
          </b>
           Medical
        </p>
        <p>
          <b>
            Notes:
          </b>
           Free health insurance. Remember to fill out paperwork again in November to reapply for free program.
        </p>
        <p />
      </div>
    </div>
  </li>
</DocumentFragment>
`;
